app/one_click_unsubscribe/rest.py

```diff
@@ -52,3 +60,57 @@ def get_unsubscribe_request_data(notification, email_address):
         "service_id": notification.service_id,
         "email_address": email_address,
     }
+
+
+def create_unsubscribe_request_reports_summary(service_id):
+    reports_summary = {"batched_reports_summaries": [], "unbatched_report_summary": {}}
```

Comment:
I’m now wondering why we are returning 2 fields here when we just join them into one list in the admin app. Would it not be better to return one list with the contents of `unbatched_report_summary` as the first item?

We can always use the `is_a_batched_report` field to tell them apart

Reply:
This would also let you move a lot of the Python logic below into a single SQL query. Not sure of the SQLAlchemy syntax but something like this would be faster and easier to follow:
```sql
select
    batch_id, count(*)
from
    unsubscribe_request
group by
    batch_id
```
Would have to figure out the `join` where `batch_id` is none but I’m sure someone will know the SQL wizardry.

Reply:
I had started off merging both batched and un-batched reports and later unpicked them to give flexibility but it is seems to be more trouble than it's worth, so I have merged them again.

Regarding the remaining logic. One slight area of complication has been that we have batched `unsubscribe_request_reports` that exist in the database and we have to present un-batched `unsubscribe_requests` as an unbatched report whose `count` and `latest_timestamp` are dynamic and potentially change every time the user views them until they are converted to a batched `unsubscribe_request_report `once the user initiates a download.
The premise I worked on was creating summaries for all reports, batched or unbatched, which are presented to the user. The summaries are not persisted in the database and are crafted by this method. Once a user initiates a download, an un-batched report summary is converted to an `unsubscribe_request_report`, (the` batch_id` it was assigned will become the batched `unsubscribe_request_report` `id` and stored in the data base.

Reply:
DId you mean `get_unbatched_unsubscribe_requests_dao` should return the `count` as well?

Reply:
I now see your point, were you referring to something like this? https://github.com/alphagov/notifications-api/blob/1c591702c3c59771bdb1d53e5902f8295f56ee33/app/dao/returned_letters_dao.py#L78